gradoop-flink/src/main/java/org/gradoop/flink/model/api/epgm/BaseGraph.java

```diff
@@ -32,7 +33,7 @@
   G extends EPGMGraphHead,
   V extends EPGMVertex,
   E extends EPGMEdge,
-  LG extends BaseGraph> {
+  LG extends BaseGraph> extends LogicalGraphLayout<G, V, E> {
```

Comment:
I think LG must extend BaseGraph<G, V, E, LG>, else the factory can construct any BaseGraph type.